files/kafka_maintenance_at_start

```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+
+CO="__consumer_offsets"
+PC="PartitionCount"
+RF="ReplicationFactor"
+KAFKA="/home/kafka"
+BIN="$KAFKA/kafka/bin"
+LOG="$KAFKA/log/shutdown.log"
+SYNC_STATUS="$KAFKA/etc/syncStatus"
+TMP_DIR="/tmp/kafka"
+ZK_URL="zookeeper.app.internal"
+ZK_PORT="2181"
+BROKERS_ALL_FILE="brokers_all_file"
+TOPICS_FILE="topics_file"
+broker_id="cat $KAFKA/etc/brokerId" # This node
+
+prepare() {
+	printf "Preparation "
+	if [ ! -d $TMP_DIR ]; then mkdir $TMP_DIR && printf "Done\n"; fi
+}
+
+first_check() {
+	printf "First check "
+	brokers_all=$(echo "ls /brokers/ids"|$BIN/zookeeper-shell.sh $ZK_URL:$ZK_PORT|tail -1| sed 's/,//g'| sed 's/\[//'| sed 's/\]//' 2>/dev/null > $TMP_DIR/$BROKERS_ALL_FILE)
+	topics_description=$($BIN/kafka-topics.sh --describe --zookeeper $ZK_URL:$ZK_PORT |grep -v $CO 2>/dev/null > $TMP_DIR/$TOPICS_FILE)
+	if [[ -f $TMP_DIR/$BROKERS_ALL_FILE ]] && [[ -f $TMP_DIR/$TOPICS_FILE ]]; then
+		if ! [[ -s $TMP_DIR/$TOPICS_FILE ]]; then
+			printf "Nothing to replicate. Exit\n"
+			echo "READY" > $SYNC_STATUS
+			exit
+		fi
+		read -a brokers <<< $(cat $TMP_DIR/$BROKERS_ALL_FILE)
+		printf "Done\n"
+	else
+		echo "Files don't exist"
+		sleep 1s
+		first_check
+	fi
+}
+
+collect_info() {
+	printf "Collect info "
+	numbers='[0-9]'
+	if ! [[ $(cat $TMP_DIR/$BROKERS_ALL_FILE) =~ $numbers ]] ; then
+		echo "error: Not a number"
+		echo "Is Zookeeper available? Checking again..."
+		sleep 2s
+	else
+		printf "Done\n"
+		echo "Available brokers: ${brokers[@]}"
```

Comment:
`...${brokers[@]}` - Argument mixes string and array. Use * or separate argument